consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessage/IbftCommitMessage.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibft.ibftmessage;
+
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.IbftSignedMessageData;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.IbftUnsignedCommitMessageData;
+import tech.pegasys.pantheon.ethereum.p2p.NetworkMemoryPool;
+import tech.pegasys.pantheon.ethereum.p2p.api.MessageData;
+import tech.pegasys.pantheon.ethereum.rlp.RLP;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import io.netty.buffer.ByteBuf;
+
+public class IbftCommitMessage extends AbstractIbftMessage {
+
+  private static final int MESSAGE_CODE = IbftV2.COMMIT;
+
+  private IbftCommitMessage(final ByteBuf data) {
+    super(data);
+  }
+
+  public static IbftCommitMessage fromMessage(final MessageData message) {
+    if (message instanceof IbftCommitMessage) {
```

Comment:
This logic is duplicated in each Ibft message, be nice to somehow move most of the common logic into a helper function in the abstract class.

Reply:
I'll see what I can do

Reply:
Done as discussed